src/Framework/Performance/Measurement.cxx

```diff
@@ -0,0 +1,28 @@
+#include "Framework/Performance/Measurement.h"
+
+#include <chrono>
+
+ClassImp(framework::performance::Measurement);
+
+namespace framework::performance {
+
+static long unsigned int get_current_time() {
+  using std::chrono::system_clock;
+  const auto current_time{system_clock::now()};
+  return system_clock::to_time_t(current_time);
```

Comment:
The reason is that you'd need to measure time deltas relative to some original time. I don't want to pre-define an original time for anyone since it may change depending on use case.

The biggest example of this is the fact that new branches are allocated and initialized on their first `add`. Perhaps you set your origin time to the 10'th event to avoid this overhead polluting your measurement.

Reply:
I think what I would find useful is the time delta between two events, i.e. between a begin_X and end_X

Reply:
Ah that makes sense, how would other measurements fit in? Would we take to measurements of memory/cpu usage for example?

Reply:
That I'm less sure about but I think runtime and resource usage are quite different kinds of measurements so recording them in the same place is probably going to make things messy/complicated _somewhere_. 

For a memory usage measurement for example, would you really care about the memory use before/after operations? Or would you want some more uniform sampling? Or some average over an operation? 

I'm much more familiar with runtime measurements so I might be wrong!